### PR TITLE
Separate out `@agoric/telemetry` in preparation for new features

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

`@agoric/telemetry` is a new package to interpret `OTEL_*` environment variables for packages that use the https://opentelemetry.io metrics provider.

### Documentation Considerations

Global variables being used are already documented in `docs/env.md`.

### CODE REVIEW

File: packages/telemetry/shim.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    if (value !== undefined) {
+      config.set(name, value);
+    }
+  });
+  return config;
+};
+
+// This is named getMetricProvider instead of getMeterProvider to avoid horrible
+// confusion with SwingSet metering.
+globalThis.getMetricProvider = (console, env) => {
```
Comment (kriskowal) [inline packages/telemetry/shim.js]: I’ve got a strong sense that the thing we want on `globalThis` should just be a `metrics`, `meter`, or `telemetry` namespace that closes over whatever `console` or `process.env` was present at time of initialization and that `getMetricsProvider` should be hidden by a thin layer of shim.

File: packages/telemetry/shim.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    const host =
+      config.get('OTEL_EXPORTER_PROMETHEUS_HOST') ||
+      PrometheusExporter.DEFAULT_OPTIONS.host ||
+      '0.0.0.0';
+
+    const port =
+      Number(config.get('OTEL_EXPORTER_PROMETHEUS_PORT')) ||
+      PrometheusExporter.DEFAULT_OPTIONS.port;
```
Comment (kriskowal) [inline packages/telemetry/shim.js]: Might validate. `Number` can return `NaN`. Domain includes hex, if that’s intended to be supported. Not sure whether `0` should silently fall through to default it’s provided explicitly.

File: packages/cosmic-swingset/src/kernel-stats.js [hunk 3]
```diff
@@ -263,36 +263,13 @@ export function exportKernelStats({
...
-      startServer: true,
-      host,
-      port,
-    },
-    () => {
-      console.log(
-        `Prometheus scrape endpoint: http://${host}:${port}${PrometheusExporter.DEFAULT_OPTIONS.endpoint}`,
-      );
-    },
+export function getMetricProvider(console, env) {
```
Comment (kriskowal) [inline packages/cosmic-swingset/src/kernel-stats.js]: If not `meter`, I think I’d favor `metrics` over `metric`.

**Review by kriskowal — CHANGES_REQUESTED** [review]
Only blocker is I’d like to be convinced that the shim needs to expose `globalThis.getMetricsProvider` over `globalThis.telemetry`. Does the end-user need to make telemetry API’s or does the user need a telemetry API?  If they need multiple, can we direct them to use the underlying module instead?

**Comment by kriskowal** [comment]
I’m at this point comfortable leaning on the convention that the shim depends on `console` and `process.env` to be present.

File: packages/cosmic-swingset/src/kernel-stats.js (hunk 3 above)
Comment (michaelfig) [reply packages/cosmic-swingset/src/kernel-stats.js]: Metrics it is.

File: packages/telemetry/shim.js (hunk 2 above)
Comment (michaelfig) [reply packages/telemetry/shim.js]: Updated to use `parseInt(..., 10)`.

File: packages/telemetry/shim.js (hunk 1 above)
Comment (michaelfig) [reply packages/telemetry/shim.js]: Eliminated use of `globalThis`.  It was misguided.

File: packages/telemetry/src/index.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  const port =
+    parseInt(OTEL_EXPORTER_PROMETHEUS_PORT || '', 10) ||
+    PrometheusExporter.DEFAULT_OPTIONS.port;
+
+  const exporter = new PrometheusExporter(
+    {
+      port,
+    },
+    () => {
+      console.log(
```
Comment (kriskowal) [inline packages/telemetry/src/index.js]: Might be better shunted to stderr in case stdout has data that can be piped.

**Review by kriskowal — APPROVED** [review]

File: packages/telemetry/src/index.js (hunk 4 above)
Comment (michaelfig) [reply packages/telemetry/src/index.js]: Hoist on my own petard.

Comment (kriskowal) [reply packages/telemetry/src/index.js:34]: I thought they were called “flag standards”. What a language.

**Pull request merged**